File: clients/java1.8_client/client/src/main/java/core/MyPassenger.java
package core;

import core.API.Elevator;
import core.API.Passenger;

public class MyPassenger {
    private Passenger passenger;

    public MyPassenger(Passenger passenger) {
        this.passenger = passenger;
    }

    public Integer getId() {
        return passenger.getId();
    }

    public Integer getElevator() {
        return passenger.getElevator();
    }

    public Integer getFromFloor() {
        return passenger.getFromFloor();
    }

    public Integer getDestFloor() {
        return passenger.getDestFloor();
    }

    public PassengerState getState() {
        return PassengerState.parse(passenger.getState());
    }

    public Integer getTimeToAway() {
        return passenger.getTimeToAway();
    }

    public String getType() {
        return passenger.getType();
    }

    public Integer getFloor() {
        return passenger.getFloor();
    }

    public Double getX() {
        return passenger.getX();
    }

    public Double getY() {
        return passenger.getY();
    }

    public Double getWeight() {
        return passenger.getWeight();
    }

    public Passenger getPassenger() {
        return passenger;
    }

    public void setElevator(Elevator elevator) {
        passenger.setElevator(elevator);
    }
}

File: clients/java1.8_client/client/src/main/java/core/MyPassenger.kt
package core

import core.API.Passenger

class MyPassenger(private val passenger: Passenger, val isMy: Boolean) {

    val id: Int?
        get() = passenger.id

    val elevator: Int?
        get() = passenger.elevator

    val fromFloor: Int
        get() = passenger.fromFloor!!

    val destFloor: Int
        get() = passenger.destFloor!!

    val state: PassengerState
        get() = PassengerState.parse(passenger.state)!!

    val timeToAway: Int?
        get() = passenger.timeToAway

    val type: String
        get() = passenger.type

    val floor: Int
        get() = passenger.floor!!

    val x: Double?
        get() = passenger.x

    val y: Double?
        get() = passenger.y

    val weight: Double?
        get() = passenger.weight

    val score: Int
        get() = if (isMy) 10 else 20

    fun setElevator(elevator: MyElevator) {
        passenger.setElevator(elevator.elevator)
    }
}
